packages/cadl-ranch-specs/http/azure/resource-manager/models/resources/singleton.tsp

```diff
@@ -0,0 +1,172 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.Core;
+using Azure.ResourceManager;
+using TypeSpec.OpenAPI;
+
+namespace Azure.ResourceManager.Models.Resources;
+
+@resource("singletonTrackedResources")
+model SingletonTrackedResource is TrackedResource<SingletonTrackedResourceProperties> {
+  @key("singletonTrackedResourceName")
+  @path
+  @segment("singletonTrackedResources")
+  @doc("arm resource name for path")
+  @pattern("^[A-Za-z0-9]([A-Za-z0-9-_.]{0,62}[A-Za-z0-9])?$")
+  name: string;
+}
+
+@doc("Singleton Arm Resource Properties.")
+model SingletonTrackedResourceProperties {
+  @visibility("read")
+  @doc("The status of the last operation.")
+  provisioningState?: ProvisioningState;
+
+  @doc("The description of the resource.")
+  description?: string;
+}
+
+@armResourceOperations
+interface SingletonTrackedResources {
+  @scenario
+  @scenarioDoc("""
+    Resource GET operation.
+    Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Models.Resources/singletonTrackedResources/singleton
```

Comment:
If you use `@singleton`, the last segment should be `/default` instead of `/singleton`. Please verify this (e.g. in playground).

Reply:
Fixed in the new version.